src/Migrations.php

```diff
@@ -288,6 +288,20 @@ public function getManager($config = null)
 
             $this->manager = new CakeManager($config, $this->output);
         } elseif ($config !== null) {
+            $defaultEnvironment = $config->getEnvironment('default');
+            try {
+                $environment = $this->manager->getEnvironment('default');
+                $oldConfig = $environment->getOptions();
+                unset($oldConfig['connection']);
+                if ($oldConfig == $defaultEnvironment) {
```

Comment:
I don't think the order of the two configs should matter. In this instance we are only concerned if the two configs have the same values, rather than having the same ordering (http://php.net/manual/en/language.operators.array.php)